packages/ssi-types/src/mapper/credential-mapper.ts

```diff
@@ -293,6 +293,82 @@ export class CredentialMapper {
     return '@context' in credential && ((credential as ICredential).type?.includes('VerifiableCredential') || false)
   }
 
+  public static isCredential(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string') {
+        if (this.isJwtEncoded(original)) {
+          const vc: IVerifiableCredential = this.toUniformCredential(original)
+          return this.isW3cCredential(vc)
+        } else if (this.isSdJwtEncoded(original)) {
+          // find out how we can differentiate between vc and vp for sd-jwts
+          return true
+        } else {
+          document = JSON.parse(original)
+        }
+      }
+      return (
+        this.isW3cCredential(document as ICredential) ||
+        this.isSdJwtDecodedCredentialPayload(document as ICredential) ||
+        this.isJwtDecodedCredential(document as OriginalVerifiableCredential) ||
+        this.isSdJwtDecodedCredential(document as OriginalVerifiableCredential)
+      )
+    } catch (e) {
+      return false
+    }
+  }
+
+  public static isPresentation(original: OriginalVerifiableCredential | OriginalVerifiablePresentation): boolean {
+    let document = original
+    try {
+      if (typeof original === 'string')
+        if (this.isJwtEncoded(original)) {
+          const vp: IVerifiablePresentation = this.toUniformPresentation(original)
+          return this.isW3cPresentation(vp)
+        } else if (this.isSdJwtEncoded(original)) {
+          return true
+        } else {
+          document = JSON.parse(original)
```

Comment:
This is a really confusing construct. You are constantly returning in the above branches, but now you have an else, and then outside of the else you start returning again. Just remove the else, or return from that else

Reply:
changed this structure. now every function in our credential-mapper will receive a jwt string/sd_jwt string/ json document